code/modules/research/xenobiology/crossbreeding/loyal.dm

```diff
@@ -0,0 +1,745 @@
+				if(rarm.heal_damage(1,1))
+					attached_user.update_damage_overlays()
+			if(ITEM_SLOT_FEET)
+				var/obj/item/bodypart/lleg = attached_user.get_bodypart(BODY_ZONE_L_LEG) //Pretty sure shoes can't be worn without both legs
+				var/obj/item/bodypart/rleg = attached_user.get_bodypart(BODY_ZONE_R_LEG) //So it's safe to assume they both exist
+				if(lleg.heal_damage(1,1))
+					attached_user.update_damage_overlays()
+				if(rleg.heal_damage(1,1))
+					attached_user.update_damage_overlays()
+			if(ITEM_SLOT_EARS)
+				var/obj/item/organ/ears/ears = attached_user.getorganslot(ORGAN_SLOT_EARS)
+				if(!ears)
+					return
+				ears.deaf = max(ears.deaf - 0.5, 0) //Can cure deafness
+				ears.damage = max(ears.damage - 0.05, 0)
+			if(ITEM_SLOT_EYES)
+				var/obj/item/organ/eyes/eyes = attached_user.getorganslot(ORGAN_SLOT_EYES)
+				if(!eyes)
+					return
+				eyes.applyOrganDamage(-1)
+				if(HAS_TRAIT_FROM(attached_user, TRAIT_BLIND, EYE_DAMAGE)) //This is just copied from oculine code for healing eye afflictions
+					if(prob(20))
+						to_chat(attached_user, "<span class='warning'>Your vision slowly returns...</span>")
+						attached_user.cure_blind(EYE_DAMAGE)
+						attached_user.cure_nearsighted(EYE_DAMAGE)
+						attached_user.blur_eyes(35)
+				else if(HAS_TRAIT_FROM(attached_user, TRAIT_NEARSIGHT, EYE_DAMAGE))
+					to_chat(attached_user, "<span class='warning'>The blackness in your peripheral vision fades.</span>")
+					attached_user.cure_nearsighted(EYE_DAMAGE)
+					attached_user.blur_eyes(10)
+				else if(attached_user.eye_blind || attached_user.eye_blurry)
+					attached_user.set_blindness(0)
+					attached_user.set_blurriness(0)
+
+/obj/item/slimecross/loyal/blue
+	colour = "blue"
+	effect_desc = "Makes an item self-cleaning."
+	loyaleffect = /datum/component/loyal_effect/blue
+
+/datum/component/loyal_effect/blue
+	prefix = "clean"
+
+/datum/component/loyal_effect/blue/Initialize(obj/item/attached_item)
+	. = ..()
+	START_PROCESSING(SSobj,src)
+
+/datum/component/loyal_effect/blue/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	return ..()
+
+/datum/component/loyal_effect/blue/process()
+	SEND_SIGNAL(attached, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_GOD) //The CLEANEST
+
+/obj/item/slimecross/loyal/metal
+	colour = "metal"
+	effect_desc = "Makes an item more resistance to brute attacks."
+	loyaleffect = /datum/component/loyal_effect/metal
+
+/datum/component/loyal_effect/metal
+	prefix = "metallic"
+
+/datum/component/loyal_effect/metal/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.armor.attachArmor(list("melee" = 10, "bullet" = 5, "bomb" = 5))
+
+/obj/item/slimecross/loyal/yellow
+	colour = "yellow"
+	effect_desc = "Makes an item glow."
+	loyaleffect = /datum/component/loyal_effect/yellow
+
+/datum/component/loyal_effect/yellow
+	prefix = "glowing"
+
+/datum/component/loyal_effect/yellow/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.set_light(l_range = 5, l_power = 1) //A bit wider than a flashlight
```

Comment:
This acts funkily with items  that already emit light, like flashlights. How would I change this?